### PR TITLE
Applied css to integration-endpoint screen.

### PR DESCRIPTION
**Pull request by SezginEge**

### CODE REVIEW

**Comment by gokhangulbiz** [comment]
@daronyondem Could you please review style implementations?

**Comment by SezginEge** [comment]
@daronyondem This pr needs to clean up. But, to follow changes on pr, i didn't do rebase. Let me know when you finished review.

**Comment by daronyondem** [comment]
The "Create" buttons right border is not aligned to the right end of the gray box surrounding text input controls. 
We need to decrease the amount of space between the horizontal gray line between and the page header, create button.
Same applies for the delete button as well.
Can we make the dropdowns expend according to their content instead of 100% enlargement.

**Comment by SezginEge** [comment]
@daronyondem Updated.

**Comment by daronyondem** [comment]
Thanks for the updates.

The first time I opened the page I had the "Required fields" error on my page. I prefer not having the error before I click the "Create" button for the first time. Can we fix that?

screenshot at may 17 11-41-37

I don't like how this looks. Can we try pushing the dropdown underneath the label. We need to make it look like the textboxes. As far as I can see textboxes have their shadows inwards where the dropdown has it outwards. In addition to that, can we change how the dropdown's full list style looks? When I click the dropdown the list that pops up looks like a Win32 list on top of a modern UI. Feels weird.

**Comment by SezginEge** [comment]
@daronyondem I updated the pr.

**Comment by daronyondem** [comment]
Thanks.

The column name "Management Portal URL" needs to change to "Management Portal Access". This is the same type of change we implement in MSG integration page as well. See https://github.com/SezginEge/xomni-cms-vnext/commit/8f56aedb2903ee64c755cda5650cedcaf4b7757d

Status "Succeeded" needs to be changed to "Enabled". By the way, do we have a different status? What happens if the status is not succeeded? Does the user need to Disable and Enable again to retry?

When I click the "Management Portal Link" it asks me to Sign In. Shouldn't I already be signed in?

When I click "Delete" and I have MSG Integration enabled is it going to warn me about it?

**Comment by SezginEge** [comment]
Okay. I will change column name and add warn screen.

There are three different status; InProgress, Succeeded, Failed. By default, If user enable the msg-integration, API's return InProgress status and it will take half an hour to be succeeded. Are we check the status for half an hour?

If the status is not succeed, user need to disable to enable again. But, i will hande that in background.

**Comment by ilkayilknur** [comment]
@SezginEge How will we proceed in this PR ? As far as I remember, we will accept the Succeeded state and creation state into the dev branch. But PR still looks unmergable. Any updates ?

**Comment by SezginEge** [comment]
@ilkayilknur I didn't update yet this PR. Please be patient. I will add confirm box to integration end point and send new PR.

**Comment by daronyondem** [comment]
@ilkayilknur We had a discussion with Sezgin about this PR back in the office. He is going to provide a way to reproduce all QA scenarios with reproduction steps. We have additional expectations, as in a "InProgress" screen and a "Error" state within the initial "setup" screen. @SezginEge should have already defined these steps in his PBI. We agreed on him submitting the initial process as a PR and next week push the additional functionalities to improve UX as a separated PR. 

@SezginEge Please focus on the functional when you send comments here. We all know that you did not update this PR yet. That is the why @ilkayilknur is asking for an update. Why would you state the obvious? Additional, why are you suggesting that @ilkayilknur is impatient? He is just asking for an update about a PR that is not updated for two days and you could have just explained the situation as I did in my comment above. I expect you to craft your attitude to focus on functional deliveries. 

Btw, just fyi, in case you missed https://github.com/XomniCloud/xomni-cms-vnext/pull/52 The confirm box PR is merged. You don't need to add it into your PR. Just a regular pull will suffice.

**Comment by SezginEge** [comment]
@daronyondem I got it. 

@daronyondem  @ilkayilknur Updated the PR.

File: src/XOMNI.CMS.FrontEnd/src/pages/management/integration-endpoint-page/integration-endpoint.ts [hunk 1]
```diff
@@ -64,17 +85,33 @@ export class viewModel extends cms.infrastructure.baseViewModel {
...
+        };
+        this.showDialog(content);
     }
 
     showNoDataFoundDialog() {
         this.showCustomErrorDialog("No data found for selected dates.");
     }
+}
+
+export enum IntegrationEndpointStatusType {
```
Comment (ilkayilknur) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/integration-endpoint-page/integration-endpoint.ts]: Why did we duplicate this enum ? Our TS SDK already contains same enum https://github.com/XomniCloud/xomni-sdk-ts-preview/blob/master/xomni-sdk-ts-preview/src/models/management/integration/endpoint-status-type.ts
Can you delete the duplicated enum and update the PR again ?
Thanks.

File: src/XOMNI.CMS.FrontEnd/src/pages/management/integration-endpoint-page/integration-endpoint.ts [hunk 2]
```diff
@@ -64,17 +85,33 @@ export class viewModel extends cms.infrastructure.baseViewModel {
...
-            (e) => {
-                this.showErrorDialog();
-            }
-            );
+            DataContext: this
+        };
+        this.showDialog(content);
     }
 
     showNoDataFoundDialog() {
```
Comment (ilkayilknur) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/integration-endpoint-page/integration-endpoint.ts]: It looks like this is an unused method in this page. Could you delete the method and update PR ?
Thanks.

Comment (SezginEge) [inline src/XOMNI.CMS.FrontEnd/src/pages/management/integration-endpoint-page/integration-endpoint.ts]: @ilkayilknur I removed unused method in the page.

**Comment by SezginEge** [comment]
@ilkayilknur I removed unused method in the page.

**Comment by SezginEge** [comment]
@ilkayilknur Updated. Removed unused method. Also, I added clearInputFields method for clean input fields after the delete operation.

**Comment by ilkayilknur** [comment]
@SezginEge Thanks Sezgin. LGTM :+1: 
Can you take a look @daronyondem for UX Approval ?

**Comment by daronyondem** [comment]
How can I test the other scenarios where the status needs to be something other than "Enabled"?

When I disable the Integration it is throwing out multiple errors consecutively. See screen capture below:

manyerrors

When I enable a tenant I always get "xomniapimanagementtestcode" as the service name instead of what I provide. Is this expected?

**Comment by ilkayilknur** [comment]
@daronyondem thanks for reporting. We fixed the bug in https://github.com/XomniCloud/xomni-cms-vnext/pull/56
Getting the same service name after disable-enable scenario is expected. We don't remove API management instance when delete API is called. We keep a reference of service and if tenant enables endpoint integration again, we give  the same API management instance.
Please let me know when you test enabled and disabled states. I can delete API management service from backend and you can test InProgress state by enabling integration endpoint again.

**Comment by daronyondem** [comment]
Thanks

**Pull request closed without merging**